src/Adapters/Auth/apple.js

```diff
@@ -21,7 +21,11 @@ const getApplePublicKey = async () => {
     throw e;
   }
 
-  const key = data.keys[0];
+  const key = data.keys.find(key => key.kid === keyId);
+
+  if (!key) {
+    throw Error('Public key with matching key ID to token not found');
+  }
```

Comment:
We currently have the keys cached (data). If there is no matching key we should attempt to fetch new keys.

Reply:
ok, to reflect this I now check the cached key to see if it matches key ID first thing.  If it matches it returns, if not it invokes jwks-rsa.  I also have the cache enabled for jwks-rsa so it will have the last 5 keys cached for an hour and return without an http request if it has the matching key.  I am not longer returning the cached key from the catch block of the request for signing key because if we've gotten that far then the key id of the cached key didn't match the requested one